src/server/auth/server/oidc.go

```diff
@@ -464,30 +395,19 @@ func (a *apiServer) handleOIDCExchangeInternal(ctx context.Context, sp *Internal
 		return "", "", errors.Wrapf(err, "failed to exchange code")
 	}
 
-	var verifier = sp.Provider.Verifier(&oidc.Config{ClientID: conf.ClientID})
 	// Extract the ID Token from OAuth2 token.
 	rawIDToken, ok := tok.Extra("id_token").(string)
 	if !ok {
 		return "", "", errors.New("missing id token")
 	}
 
 	// Parse and verify ID Token payload.
-	idToken, err := verifier.Verify(ctx, rawIDToken)
+	idToken, claims, err := sp.validateIDToken(ctx, rawIDToken)
 	if err != nil {
 		return "", "", errors.Wrapf(err, "could not verify token")
 	}
 
-	// Use the ID token passed from the authorization callback as our token source
-	ts := oauth2.StaticTokenSource(
-		&oauth2.Token{
-			AccessToken: tok.AccessToken,
-		},
-	)
-	userInfo, err := sp.Provider.UserInfo(ctx, ts)
-	if err != nil {
-		return "", "", errors.Wrapf(err, "could not get user info")
-	}
-	return idToken.Nonce, userInfo.Email, nil
+	return idToken.Nonce, claims.Email, nil
```

Comment:
This is just me not understanding the ID Token claims, but do we need to check EmailVerified here too?